worker/src/monitor/metrics_collector.rs

```diff
@@ -0,0 +1,67 @@
+use std::{thread::sleep, time::Duration};
+
+use proto::common::node::Metrics;
+use sysinfo::System;
+
+const CPU_DELAY_IN_MILLIS: u64 = 500;
+
+/// Struct containing system's information such as CPU and Memory.
+pub struct MetricsCollector {
+    system: System,
+}
+impl MetricsCollector {
+    /// Instantiates a new `MetricsCollector`.
+    ///
+    /// Responsable for collecting metrics about the System, such as
+    /// CPU Percentage Usage, Free Memory Space.
+    pub fn new() -> Self {
+        Self {
+            system: System::new_all(),
+        }
+    }
+
+    /// Get the `cpu_usage` and `free_memory` metrics from the current system.
+    ///
+    /// When calling this method, will sleep the thread for
+    /// `CPU_DELAY_IN_MILLIS`.
+    pub fn get_metrics(&mut self) -> Metrics {
+        Metrics {
+            cpu_usage: self.get_cpu_usage(),
+            mem_total_mib: self.get_total_memory(),
+            mem_used_mib: self.get_used_memory(),
+        }
+    }
+
+    fn get_cpu_usage(&mut self) -> Vec<f64> {
```

Comment:
Does not return a single usage value 

```suggestion
    fn get_cpus_usage(&mut self) -> Vec<f64> {
```